src/Caching/SqlServer/src/MonoDatabaseOperations.cs

```diff
@@ -98,12 +79,9 @@ protected override async Task<byte[]> GetCacheItemAsync(string key, bool include
             }
 
             byte[] value = null;
-            TimeSpan? slidingExpiration = null;
-            DateTimeOffset? absoluteExpiration = null;
-            DateTimeOffset expirationTime;
             using (var connection = new SqlConnection(ConnectionString))
             {
-                var command = new SqlCommand(SqlQueries.GetCacheItem, connection);
+                var command = new SqlCommand(query, connection);
```

Comment:
Do we have some sort of test for this? I'm surprised this bug snuck through... it seems like a good place to test.

Reply:
Good point. Not so easy. Let me explain: `protected DatabaseOperations.GetCacheItem` (Async and not Async) functions are called only by public `DatabaseOperations.GetCacheItem[Async]` that makes the call passing the parameter `includeValue` to true, and `DatabaseOperations.RefreshCacheItem[Async]` which sets the parameter `includeValue` to false (as doesn’t need the value). 

When the parameter `includeValue` is true GetCacheItem opens a dataReader with two commands in the sql string: one to update the expiration date and another one to get and return the cached value (or null if not found). When the parameter `includeValue` is false, only opens the dataReader with a single update command in the sql string to update the expiration date and always returns null. 

It is difficult to test the value of the sql string inside the function or the commands executed in the database depending on the value of the parameter. 

In my opinion, the cleanest approach is not related to testing, but to refactor a bit the code. Roughly explained, I would get rid of the boolean parameter `includeValue` and let the function `GetCacheItemAsync` just with the implementation for refresh the expiration date and get the value. Next, implement the `RefreshCacheItem` into another method/function exclusively written for that. 

Then each method will use the proper command string and implementation without any conditional. It is not a so complex refactoring.

Given the “not so high” priority in this project and the limited impact in performance and features (this is more housekeeping), I would vote to make it in another pull request when there is time to do it.

Sorry for this loong explanation but I couldn't make it shorter to propose another approach instead of modifying the tests.

Reply:
ok, thanks for the explanation.